lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -704,6 +712,17 @@ def get_docker_environment(env, env_files):
     return sorted(env_list)
 
 
+def get_nanoseconds_from_raw_option(value):
+    if value is None:
+        return None
+    elif isinstance(value, int):
+        return value
+    elif isinstance(value, string_types):
+        return convert_duration_to_nanosecond(value)
+    else:
+        raise ValueError
```

Comment:
Maybe also add the name of the option as an argument, so you can compose a helpful error message here.